design-proposals/mass_machine_type_transition.md

```diff
@@ -0,0 +1,208 @@
+
+The options to select a specific namespace or restart the VMs immediately are configurable with the `NAMESPACE` and `RESTART_NOW` environment variables respectively.
+
+Given a VM with the following spec:
+````
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+...
+spec:
+  running: false
+  template:
+    spec:
+      domain:
+        machine:
+          type: pc-q35-rhel8.2.0
+````
+When the MMTT is invoked, the VM spec will be updated to: 
+````
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+...
+spec:
+  running: false
+  template:
+    spec:
+      domain:
+        machine:
+          type: pc-q35-rhel9.2.0
+````
+where `pc-q35-rhel9.2.0` is the latest machine type version.
+
+Along with machine types being in the format `pc-q35-rhelx.x.x`, a VM may have a machine type of `q35`, an alias for the current latest machine type version *at the time the VM is started*.
+
+For example, this VM:
+````
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+...
+spec:
+  running: true
+  template:
+    spec:
+      domain:
+        machine:
+          type: q35
+````
+may have been started when a RHEL-8 machine type was the latest machine type version and not been stopped or restarted since. Its VMI would thus look like this:
+````
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+...
+spec:
+  domain:
+    machine:
+      type: q35
+status:
+  machine:
+    type: pc-q35-rhel8.2.0
+````
+and need to have its machine type updated.
+
+To handle this, any running VMs with the machine type `q35` will be labelled with `restart-vm-required`:
+````
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+	labels: "restart-vm-required"="true"
+spec:
+  running: true
+  template:
+    spec:
+      domain:
+        machine:
+          type: q35
+````
+Once the VM has been stopped or restarted, the `restart-vm-required` label will be removed:
+````
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+	labels:
+spec:
+  running: false
+  template:
+    spec:
+      domain:
+        machine:
+          type: q35
+````
+Now when a new VMI is created when starting this VM, it should have the latest machine type version:
+````
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+...
+spec:
+  domain:
+    machine:
+      type: q35
+status:
+  machine:
+    type: pc-q35-rhel9.2.0
+````
+
+## Update/Rollback Compatibility
+As both the minimum supported machine type version and the latest machine type version change in the future, these values are global constants in the MMTT package. As the versions are updated in the future, these global constants can also be updated accordingly, allowing for easy maintainability.
```

Comment:
Ah so these are going to be constants? Could you reference this aspect in the design section?

Reply:
Yes these will both be constants, I've updated the design section.

Reply:
As said above I'd prefer dynamic values over constant ones